org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/LoginWindow.java

```diff
@@ -92,36 +103,125 @@ public void init(IWebClient app)
     	this.ctx = Env.getCtx();
         this.app = app;
         initComponents();
-        this.appendChild(pnlLogin);
+		if (pnlLogin != null)
+			this.appendChild(pnlLogin);
         this.setStyle("background-color: transparent");
         // add listener on 'ENTER' key for the login window
         addEventListener(Events.ON_OK,this);
 		this.addEventListener(ON_DEFER_LOGOUT, this);
         setWidgetListener("onOK", "zAu.cmd0.showBusy(null)");
     }
 
-    private void initComponents()
-    {
-        createLoginPanel();
-    }
+	private void initComponents()
+	{
+		Object result = getDesktop().getSession().getAttribute(ISSOPrinciple.SSO_PRINCIPLE_SESSION_NAME);
+		if (result == null)
+		{
+			createLoginPanel();
+		}
+		else
+		{
+			ssoLogin(result);
+		}
+	}
+
+	/**
+	 * Show role panel after SSO authentication.
+	 * 
+	 * @param result session principle to get user and language.
+	 */
+	private void ssoLogin(Object result)
```

Comment:
if result is a "Principal" entity, should just name it as principal, same for the initComponents above as well.